internal/test/framework/windowsvm.go

```diff
@@ -157,21 +158,20 @@ func (w *windowsVM) CopyFile(filePath, remoteDir string) error {
 	return nil
 }
 
-// RetrieveFiles retrieves list of files from remote directory to the local directory.
-// The implementation can be changed if the use-case arises. As of now, we're doing a best effort
-// to collect every log possible. If a retrieval of file fails, we would proceed with retrieval
-// of other log files.
-func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
+// RetrieveDirectories retrieves a whole directory recursively from the remote windows VM to the local/CI directory.
+// This function recursively calls itself if it finds a directory inside, and it calls the retrieveFiles function
+// if it encounters a file.
+func (w *windowsVM) RetrieveDirectories(remoteDir string, localDir string) error {
+
 	if w.sshClient == nil {
-		return fmt.Errorf("RetrieveFile cannot be called without a ssh client")
+		return fmt.Errorf("cannot retrieve remote directory without a ssh client")
 	}
 
-	// Create local dir
+	// Create a local dir
```

Comment:
`Create a local dir` that is whats happening, but the line below is self explanatory
I suggest saying what the directory is for :)